finality-aleph/src/finalization.rs

```diff
@@ -26,7 +26,7 @@ where
     C: HeaderBackend<B> + LockImportRun<B, BE> + Finalizer<B, BE>,
 {
     client: Arc<C>,
-    metrics: AllBlockMetrics,
+    metrics: RefCell<AllBlockMetrics>,
```

Comment:
why refcell?

Reply:
Otherwise I would have to change `finalize_block` in `trait BlockFinalizer` to accept `&mut self` instead `&self`.
This enforces changing traits in `block::substrate::finalizer` and `finalization` to accept `&mut self` instead of `&self`. I didn't want to change these traits if I didn't have to, but we can consider - what would be your opinion on that?

Reply:
Metrics are Clone, so can't we simply clone them instead of storing them in a RefCell? All prometheus primitives (Counter, Histogram etc) are Arcs so they can be safely cloned.

Reply:
The problem is the `imported_cache` which we want to mutate - cloning whole metrics won't be good. Tho I see that in `TimingBlockMetrics` the cache you implemented is under `Arc<Mutex<...>>`. I can do it the same way :+1:

Reply:
Arc is fine, it is a solution that we use in the project.